packages/aws-cdk/test/api/logs/cloudwatch-logs.test.ts

```diff
@@ -0,0 +1,287 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { CloudFormation } from 'aws-sdk';
+import { registerCloudWatchLogGroups } from '../../../lib/api/logs/cloudwatch-logs';
+import { testStack, TestStackArtifact } from '../../util';
+import { MockSdkProvider } from '../../util/mock-sdk';
+
+let logsMockSdkProvider: LogsMockSdkProvider;
+let mockGetEndpointSuffix: () => string;
+
+beforeEach(() => {
+  jest.useFakeTimers();
+  logsMockSdkProvider = new LogsMockSdkProvider();
+  mockGetEndpointSuffix = jest.fn(() => 'amazonaws.com');
+  logsMockSdkProvider.stubGetEndpointSuffix(mockGetEndpointSuffix);
+  // clear the array
+  currentCfnStackResources.splice(0);
+});
+afterAll(() => {
+  jest.useRealTimers();
+});
+
+test('add log groups from lambda function', async () => {
+  // GIVEN
+  pushStackResourceSummaries(stackSummaryOf('Func', 'AWS::Lambda::Function', 'my-function'));
+  const cdkStackArtifact = cdkStackArtifactOf({
+    template: {
+      Resources: {
+        Func: {
+          Type: 'AWS::Lambda::Function',
+          Properties: {
+            FunctionName: 'my-function',
+          },
+        },
+      },
+    },
+  });
+  const result = await registerCloudWatchLogGroups(logsMockSdkProvider.mockSdkProvider, cdkStackArtifact);
+  expect(result.groups).toEqual(new Set(['/aws/lambda/my-function']));
+});
+
+test('add log groups from ECS Task Definitions', async () => {
+  // GIVEN
+  pushStackResourceSummaries(stackSummaryOf('Def', 'AWS::ECS::TaskDefinition', 'app'));
+  pushStackResourceSummaries(stackSummaryOf('LogGroup', 'AWS::Logs::LogGroup', 'log_group'));
+  const cdkStackArtifact = cdkStackArtifactOf({
+    template: {
+      Resources: {
+        LogGroup: {
+          Properties: {
+            Type: 'AWS::Logs::LogGroup',
+            LogGroupName: 'log_group',
+          },
+        },
+        Def: {
+          Type: 'AWS::ECS::TaskDefinition',
+          Properties: {
+            Family: 'app',
+            ContainerDefinitions: [
+              {
+                LogConfiguration: {
+                  LogDriver: 'awslogs',
+                  Options: {
+                    'awslogs-group': {
+                      Ref: 'LogGroup',
+                    },
```

Comment:
Let's try to preserve vertical space where we can:

```suggestion
                    'awslogs-group': { Ref: 'LogGroup' },
```